internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -404,7 +404,7 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 			}
 
 			return len(machineSets.Items)
-		}, timeout*5).Should(BeEquivalentTo(0))
+		}, timeout*10).Should(BeEquivalentTo(0))
```

Comment:
Can we try to disable the List caching behavior and see if that solves the issue?

Reply:
I don't think this is down to caching - I think the cause is that some times the MachineSets take a long time to go away. I don't think I've seen the cache take anything like 5 seconds to update